fix(client): hide console on JS-only interactive examples

bengguankoay opened the pull request:
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitHub Codespaces.


Closes #62796

---

Console output now appears once instead of twice in JavaScript-only examples. Console panel only visible when HTML is present.

Changes:
- Modified showConsole logic to check for both JavaScript AND HTML files
- JS-only examples now hide console panel (output shown once in preview)
- HTML+JS examples continue showing both preview (HTML) and console (logs)

The duplication occurred because layout:'console' displays console output in the preview area, while showConsole:true displays it again in the console panel. For JS-only examples, hiding the console panel eliminates the redundancy while maintaining functionality.

Before: 
<img width="1312" height="938" alt="Screenshot 2025-11-28 at 10 19 27 PM" src="https://github.com/user-attachments/assets/2a456edb-3615-4d08-a1ce-042bf684616e" />
After: 
<img width="1312" height="938" alt="Screenshot 2025-11-28 at 10 41 19 PM" src="https://github.com/user-attachments/assets/595973d5-dfd3-4e09-9dbb-b91edc635a3e" />

bengguankoay committed:
fix(curriculum): hide console panel for JS-only interactive examples

Console output now appears once instead of twice in JavaScript-only
examples. Console panel only visible when HTML is present.

huyenltnguyen commented on client/src/templates/Challenges/components/interactive-editor.tsx:
Then update this to

```suggestion
          layout: layout,
```

huyenltnguyen reviewed (commented):
Please help add a test for the JS-only case to `e2e/interactive-editor.spec.ts`.

You'll need to mock the challenge data so it returns only a JS file. You can look at the existing tests in the file to see how challenge data is mocked.

bengguankoay committed:
fix: resolve PR comment, refactored console visibility logic

bengguankoay committed:
test(e2e): resolve PR comment, add JS-only interactive editor test case

Add test coverage for interactive editor when only JavaScript file is present (no HTML). Verifies that the editor correctly renders with console layout and toggle functionality works as expected.

bengguankoay committed:
Merge remote-tracking branch 'upstream/main' into fix/closed-by-default-for-interactive-editor-in-JS-only

huyenltnguyen reviewed (commented):
Thanks for the update!

There is just minor adjustment we should make. The PR will be good to go then.

bengguankoay committed:
Merge remote-tracking branch 'upstream/main' into fix/closed-by-default-for-interactive-editor-in-JS-only

bengguankoay committed:
resolve PR comment, skip unnecessary steps in unit test

huyenltnguyen reviewed (approved):
Thanks @bengguankoay!
